tools/topology/topology2/platform/intel/sdw-amp-feedback.conf

```diff
@@ -0,0 +1,64 @@
+Object.Dai {
+	ALH."515" {
+		id 		$SDW_AMP_IN_BE_ID
+		direction	"capture"
+		name		$SDW_SPK_IN_STREAM
+		default_hw_conf_id	1
+		rate			48000
+		channels		2
+
+		Object.Base.hw_config."ALH515" {
+			id	1
+		}
+	}
+}
+
+Object.Pipeline {
+	passthrough-capture.30 {
+		index 30
+		Object.Widget.pipeline.1.stream_name	"SPK IN stream"
+
+		Object.Widget.copier.1.stream_name	"amp feedback"
+		Object.Widget.copier.1.Object.Base.audio_format.1 {
+			# 32 -> 16 bits conversion is done here,
+			# so in_bit_depth is 32 (and out_bit_depth is 16).
+			in_bit_depth	32
+		}
+	}
+	passthrough-be.31 {
+		direction	"capture"
+		index 31
+		copier_type "ALH"
+		Object.Widget.pipeline.1 {
+			stream_name	'SPK IN stream'
+		}
+		Object.Widget.copier.1 {
+			stream_name	$SDW_SPK_IN_STREAM
+			dai_type	"ALH"
+			copier_type	"ALH"
+			type		"dai_out"
+			node_type $ALH_LINK_INPUT_CLASS
+		}
+	}
+}
+
+Object.PCM {
+	pcm.3 {
+		name	"Amp feedback"
+		id 3
+		direction	"capture"
+		Object.Base.fe_dai."Amp feedback" {}
+
+		Object.PCM.pcm_caps."capture" {
+			name "amp feedback"
+			formats 'S16_LE,S32_LE'
+		}
+	}
+}
+
+Object.Base {
+	route.30 {
+		source	"copier.ALH.31.1"
+		sink	"copier.host.30.1"
```

Comment:
what happens if we want to split the feedback? This is fine for an initial start, but we do know this is going to have to be split between the dai and host parts

Reply:
> what happens if we want to split the feedback? This is fine for an initial start, but we do know this is going to have to be split between the dai and host parts

@plbossart I am not sure if I get your question. The dai and host are in different pipelines now. If we want to add some widgets between dai and host, we can add the widgets in the pipeline and add routes for them.

Reply:
you can add some macro definition like linking HOST_PIPELINE_MODULE to DAI_PIPELINE_MODULE